src/content/app/regulatory-activity-viewer/components/region-overview/region-overview-image/region-overview-gene/RegionOverviewGene.tsx

```diff
@@ -0,0 +1,227 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ScaleLinear } from 'd3';
+
+import { GENE_HEIGHT } from 'src/content/app/regulatory-activity-viewer/components/region-overview/region-overview-image/regionOverviewImageConstants';
+
+import type { GeneInTrack } from 'src/content/app/regulatory-activity-viewer/helpers/prepare-feature-tracks/prepareFeatureTracks';
+import type {
+  ExonInRegionOverview,
+  CDSFragment
+} from 'src/content/app/regulatory-activity-viewer/types/regionOverview';
+
+import styles from './RegionOverviewGene.module.css';
+
+/**
+ * The transcript diagram is visually similar to a transcript diagram in EntityViewer.
+ * However:
+ *  - Data shape is going to be different (exons instead of spliced exons; multiple cds fragments)
+ *  - Transcript might need to be cropped if it intersects a gap in the image
+ *    - Consider svg clipPath
+ */
+
+type Props = {
+  scale: ScaleLinear<number, number>;
+  gene: GeneInTrack;
+  offsetTop: number;
+  isFocused: boolean;
+  onClick: (geneId: string) => void;
+};
+
+type Intron = {
+  start: number;
+  end: number;
+};
+
+const RegionOverviewGene = (props: Props) => {
+  const { gene, scale, offsetTop, isFocused } = props;
+  const transcript = gene.data.representative_transcript;
+  const color = isFocused ? 'black' : '#0099ff'; // <-- This is our design system blue; see if it can be imported
+
+  const trackY = offsetTop;
+
+  let transcriptStart: number | null = null;
+  let transcriptEnd: number | null = null;
+  const introns: Intron[] = [];
+
+  for (let i = 1; i < transcript.exons.length; i++) {
+    const prevExon = transcript.exons[i - 1];
+    const currentExon = transcript.exons[i];
+    const start = prevExon.end + 1;
+    const end = currentExon.start - 1;
```

Comment:
Shouldn't this be the next `start` of the next exon?

Reply:
If I remember my reasoning correctly:

A transcript with a single exon won't have an intron.

<img src="https://github.com/user-attachments/assets/f1283b3d-dcd2-4d80-ac50-29602fa5baaa" width="300" />

The minimum number of exons required to have an intron between them is two. Thus, I start counting from the second exon (`let i = 1`), and look at the previous one. An intron will start 1 position after where the previous exon ended, and will end 1 position before the current exon starts:

<img src="https://github.com/user-attachments/assets/a09c69a5-f7ad-4bfa-a4af-d7a351268e77" width="400" />